extensions/amp-consent/0.1/consent-config.js

```diff
@@ -0,0 +1,219 @@
+        (deepMerge(cmpConfig || {}, inlineConfig || {}, 1));
+
+    const consents = config['consents'];
+    user().assert(consents, '%s: consents config is required', TAG);
+    user().assert(Object.keys(consents).length != 0,
+        '%s: can\'t find consent instance', TAG);
+    if (!this.isMultiSupported_) {
+      // Assert single consent instance
+      user().assert(Object.keys(consents).length <= 1,
+          '%s: only single consent instance is supported', TAG);
+      if (config['policy']) {
+        // Only respect 'default' consent policy;
+        const keys = Object.keys(config['policy']);
+        for (let i = 0; i < keys.length; i++) {
+          if (keys[i] != 'default') {
+            user().warn(TAG, 'policy %s is currently not supported' +
+              'and will be ignored', keys[i]);
+            delete config['policy'][keys[i]];
+          }
+        }
+      }
+    }
+
+    return config;
+  }
+
+  /**
+   * Read the inline config from publisher
+   * @return {?JsonObject}
+   */
+  getInlineConfig_() {
+    // All consent config within the amp-consent component. There will be only
+    // one single amp-consent allowed in page.
+    try {
+      return getChildJsonConfig(this.element_);
+    } catch (e) {
+      throw user(this.element_).createError('%s: %s', TAG, e);
+    }
+  }
+
+  /**
+   * Read and format the CMP config
+   * The returned CMP config should looks like
+   * {
+   *   "consents": {
+   *     "foo": {
+   *       "checkConsentHref": "https://fake.com",
+   *       "promptUISrc": "https://fake.com/promptUI.html"
+   *     }
+   *   }
+   * }
+   * @return {?JsonObject}
+   */
+  getCMPConfig_() {
+    if (!isExperimentOn(this.win_, 'amp-consent-v2')) {
+      return null;
+    }
+
+    const type = this.element_.getAttribute('type');
+    if (!type) {
+      return null;
+    }
+    user().assert(CMP_CONFIG[type], '%s: invalid CMP type %s', TAG, type);
+    const importConfig = CMP_CONFIG[type];
+    this.validateCMPConfig_(importConfig);
+    const constentInstance = importConfig['consentInstanceId'];
+
+    const cmpConfig = dict({
+      'consents': dict({}),
+    });
+
+    const config = Object.assign({}, importConfig);
+    delete config['consentInstanceId'];
+
+    cmpConfig['consents'][constentInstance] = config;
+    return cmpConfig;
+  }
+
+  /**
+   * Check if the CMP config is valid
+   * @param {!JsonObject} config
+   */
+  validateCMPConfig_(config) {
+    const assertValues =
+        ['consentInstanceId', 'checkConsentHref', 'promptUISrc'];
+    for (let i = 0; i < assertValues.length; i++) {
+      const attribute = assertValues[i];
+      dev().assert(config[attribute], 'CMP config must specify %s', attribute);
+    }
+  }
+}
+
+/**
+ * Expand the passed in policyConfig and generate predefined policy entires
+ * @param {!JsonObject} policyConfig
+ * @param {!JsonObject} consentConfig
+ * @return {!JsonObject}
+ */
+export function expandPolicyConfig(policyConfig, consentConfig) {
```

Comment:
nit: Can we move this near the other public exported function? I noticed on the project we tend to have public first, then private 😄

Reply:
Let me know if I am wrong.

Reply:
I think this is fine. Everything else in the file is a class?

Reply:
Awesome Thanks! 😄